packages/block-library/src/gallery/v1/edit.js

```diff
@@ -148,7 +148,7 @@ function GalleryEdit( props ) {
 				...newAttrs,
 				// Unlike images[ n ].id which is a string, always ensure the
 				// ids array contains numbers as per its attribute type.
-				ids: map( newAttrs.images, ( { id } ) => parseInt( id, 10 ) ),
+				ids: newAttrs.images.map( ( { id } ) => parseInt( id, 10 ) ),
```

Comment:
All calls of `setAttributes()` will always call `images` as arrays, so we're safe to directly call `.map()` here.